store/src/directory.rs

```diff
@@ -392,21 +812,80 @@ where
                 Ok(v) if v.is_file() => {}
                 Ok(_) => continue,
             }
-            let ttl = match xattr::get(&path, format_xattr(crate::MetadataKey::<MKT>::Ttl.to_key()))
-            {
-                Err(e) => {
-                    log::trace!("Error looking up TTL xattr for {}: {:?}", path.display(), e);
+            // skip non-metadata files if xattr is not supported
+            if !self.xattr_enabled {
```

Comment:
ultimately, we want to remove this if statement here and have something more generic as:
```rust
+fn get_ov_ttl(path: &PathBuf, md_key: &str, xattr_enabled: bool) -> Option<SystemTime> {
+    // we should only ever play with OVs, metadata is an impl detail
+    match path.as_path().extension() {
+        None => (),
+        Some(ex) => {
+            if ex == FDO_METADATA_EX {
+                return None;
+            }
+        }
+    }
+    if xattr_enabled {
+        match xattr::get(&path, format_xattr(md_key)) {
+            Err(e) => {
+                log::trace!("Error looking up TTL xattr for {}: {:?}", path.display(), e);
+                return None;
+            }
+            Ok(None) => return None,
+            Ok(Some(val)) => match ttl_from_disk(&val) {
+                Ok(ttl) => return Some(ttl),
+                Err(e) => {
+                    log::trace!("error converting ttl from disk {:?}", e);
+                    return None;
+                }
+            },
+        };
+    }
+    let path_meta = set_metadata_extension_to_path(&path)?;
+    ......
+    ......
+    None
+}
```
The same goes for the other functions doing if/else